crates/sui-types/src/message_envelope.rs

```diff
@@ -191,8 +207,49 @@ where
         self.auth_signature.epoch
     }
 
-    pub fn verify_signature(&self, committee: &Committee) -> SuiResult {
-        self.data.verify(Some(self.auth_sig().epoch))?;
+    pub fn verify_committee_sigs_only(&self, committee: &Committee) -> SuiResult
+    where
+        <T as Message>::DigestType: PartialEq,
+    {
+        self.data.verify_epoch(self.auth_sig().epoch)?;
+        self.auth_signature
+            .verify_secure(self.data(), Intent::sui_app(T::SCOPE), committee)
+    }
+}
+
+impl<T> Envelope<T, AuthoritySignInfo>
+where
+    T: Message + AuthenticatedMessage + Serialize,
+{
+    pub fn verify_signatures_authenticated(
+        &self,
+        committee: &Committee,
+        verify_params: &VerifyParams,
+    ) -> SuiResult {
+        self.data.verify_epoch(self.auth_sig().epoch)?;
+        self.data.verify_message_signature(verify_params)?;
+        self.auth_signature
+            .verify_secure(self.data(), Intent::sui_app(T::SCOPE), committee)
+    }
+
+    pub fn verify_authenticated(
+        self,
+        committee: &Committee,
+        verify_params: &VerifyParams,
+    ) -> SuiResult<VerifiedEnvelope<T, AuthoritySignInfo>> {
+        self.verify_signatures_authenticated(committee, verify_params)?;
+        Ok(VerifiedEnvelope::<T, AuthoritySignInfo>::new_from_verified(
+            self,
+        ))
+    }
+}
+
+impl<T> Envelope<T, AuthoritySignInfo>
+where
+    T: Message + UnauthenticatedMessage + Serialize,
+{
+    pub fn verify_signatures(&self, committee: &Committee) -> SuiResult {
```

Comment:
I think it's a great opportunity to make these function names less confusing - `verify_authority_signatures` and below and for `impl<T, const S: bool> Envelope<T, AuthorityQuorumSignInfo<S>>`